os/arch/arm/src/common/up_mpuinit.c

```diff
@@ -50,11 +50,119 @@
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
+static struct mpu_usages_s g_mpu_region_usages;
+
+#ifdef CONFIG_ARMV7M_MPU
+#define MPU_MAX_NREGION		CONFIG_ARMV7M_MPU_NREGIONS
+#elif defined(CONFIG_ARMV8M_MPU)
+#define MPU_MAX_NREGION		CONFIG_ARMV8M_MPU_NREGIONS
+#endif
 
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: mpu_regs_initialize
+ *
+ * Description:
+ *   Initialize the MPU region numbers.
+ *
+ ****************************************************************************/
+
+void mpu_regs_initialize(FAR struct mpu_usages_s *mpu)
+{
+
+	uint8_t offset = mpu->nregion_board_specific;
+
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
```

Comment:
`CONFIG_SUPPORT_COMMON_BINARY` and `CONFIG_OPTIMIZE_APP_RELOAD_TIME` can be enabled only when `CONFIG_APP_BINARY_SEPARATION`.
So I think you can move line 87 in the above.

Reply:
But since we need to distinguish between the scenarios of configuring complete RAM partition vs that of (text, RO & data) separately, we need CONFIG_OPTIMIZE_APP_RELOAD_TIME
And to have a common binary usage , we need CONFIG_SUPPORT_COMMON_BINARY